File: frontend/src/services/customerService.js
import axiosInstance from '../utils/axiosConfig';

export const customerService = {
  getAllCustomers: async (searchQuery = '', filters = {}) => {
    const params = new URLSearchParams();
    if (searchQuery) params.append('search', searchQuery);
    if (filters.city) params.append('city', filters.city);
    if (filters.customerType) params.append('customerType', filters.customerType);
    if (filters.fromDate) params.append('fromDate', filters.fromDate);
    if (filters.toDate) params.append('toDate', filters.toDate);
    
    const queryString = params.toString();
    const response = await axiosInstance.get(`/customers${queryString ? `?${queryString}` : ''}`);
    return response.data;
  },

  getCustomerById: async (id) => {
    const response = await axiosInstance.get(`/customers/${id}`);
    return response.data;
  },

  createCustomer: async (customerData) => {
    const response = await axiosInstance.post('/customers', customerData);
    return response.data;
  },

  updateCustomer: async (id, customerData) => {
    const response = await axiosInstance.put(`/customers/${id}`, customerData);
    return response.data;
  },

  deleteCustomer: async (id) => {
    const response = await axiosInstance.delete(`/customers/${id}`);
    return response.data;
  },

  getCustomerBookings: async (id) => {
    const response = await axiosInstance.get(`/customers/${id}/bookings`);
    return response.data;
  },

  getCustomer360View: async (id) => {
    const response = await axiosInstance.get(`/customers/${id}/360`);
    return response.data;
  },

  importCustomers: async (file) => {
    const formData = new FormData();
    formData.append('file', file);
    const response = await axiosInstance.post('/customers/import', formData, {
      headers: { 'Content-Type': 'multipart/form-data' }
    });
    return response.data;
  },

  updateCustomerPhone: async (id, phone) => {
    const response = await axiosInstance.patch(`/customers/${id}/phone`, { phone });
    return response.data;
  },

  updateCustomerEmail: async (id, email) => {
    const response = await axiosInstance.patch(`/customers/${id}/email`, { email });
    return response.data;
  }
};

File: backend/src/main/java/com/dms/demo/service/CustomerService.java
package com.dms.demo.service;

import java.time.LocalDate;
import java.util.List;

import org.springframework.data.domain.Pageable;
import org.springframework.web.multipart.MultipartFile;

import com.dms.demo.dto.request.CustomerRequest;
import com.dms.demo.dto.response.BookingResponse;
import com.dms.demo.dto.response.Customer360Response;
import com.dms.demo.dto.response.CustomerResponse;
import com.dms.demo.dto.response.PagedResponse;
import com.dms.demo.enums.CustomerType;

public interface CustomerService {
    CustomerResponse createCustomer(CustomerRequest request);
    CustomerResponse updateCustomer(Long id, CustomerRequest request);
    CustomerResponse getCustomerById(Long id);
    List<CustomerResponse> getAllCustomers();
    PagedResponse<CustomerResponse> getAllCustomersPaged(String search, Pageable pageable);
    List<CustomerResponse> searchCustomers(String keyword);
    CustomerResponse searchByPhone(String phone);
    CustomerResponse searchByEmail(String email);
    List<BookingResponse> getCustomerBookings(Long customerId);
    Customer360Response getCustomer360View(Long customerId);
    void deleteCustomer(Long id);
    List<CustomerResponse> importCustomersFromCsv(MultipartFile file);
    List<CustomerResponse> getCustomersWithFilters(String search, String city, CustomerType customerType, LocalDate fromDate, LocalDate toDate);
}
